src/v2/blobs.rs

```diff
@@ -82,7 +82,14 @@ impl Client {
                     | StatusCode::FOUND => {
                         trace!("Got moved status {:?}", r.status());
                     }
-                    _ => return Either::A(future::ok(r)),
+                    StatusCode::OK => {
```

Comment:
This closure is only meant to handle top-level redirections. This whole closure at some point should be replaced by https://github.com/camallo/dkregistry-rs/issues/44.

The error status check is correct, but I suggest not to place it here. A better place would be a dedicated `and_then` after this one and before body concatenation (or directly in that closure).

Reply:
Looking at it more closely it would be necessary to construct a loop which handles a chain of requests until it falls into a case where the final state is known. E.g. you could have multiple redirections which would slip through here too.

Reply:
Indeed. That's the point of https://github.com/camallo/dkregistry-rs/issues/44#issuecomment-430135569.

Reply:
I see. I'll give it a shot to use *reqwest* for the *blobs* module.

Reply:
I would be happy to see that, but I don't know if it is possible without reworking the whole client. If your experiment starts leaning toward to the latter, I think you'd get better value by landing just the status-checking in this PR and split the reqwest rework.